components/misc/rng.cpp

```diff
@@ -1,28 +1,24 @@
 #include "rng.hpp"
-#include <cstdlib>
-#include <ctime>
+#include <random>
 
 namespace Misc
 {
 
-    void Rng::init()
-    {
-        std::srand(static_cast<unsigned int>(std::time(NULL)));
-    }
+    std::minstd_rand Rng::generator = std::minstd_rand();
```

Comment:
Global object in multithreaded application without any synchronization if it is being used from different threads will produce race condition.

Reply:
Ouch. Could I just add a lock here?

Reply:
Yes, I think, `std::mutex` and `std::lock_guard` should be fine. But make private both generator and mutex.

Reply:
IIRC all uses of Rng are from the main thread anyway. Our threads are mostly used for deterministic loading tasks where you can't possibly need anything random. Just declare the class not thread safe and be done with it.